scripts/check_e2e_tests_are_captured_in_ci_test.py

```diff
@@ -0,0 +1,251 @@
+                        check_e2e_tests_are_captured_in_ci.main()
+
+    def test_main_with_invalid_travis_script_test_suite_length(self):
+        travis_path_swap = self.swap(
+            check_e2e_tests_are_captured_in_ci,
+            'read_and_parse_travis_yml_file',
+            self._mock_read_travis_yml_file)
+
+        mock_tests_to_remove = self._mock_tests_to_remove()
+
+        mock_e2e_travis_script = self.swap(
+            check_e2e_tests_are_captured_in_ci,
+            'get_e2e_suite_names_from_script_travis_yml_file',
+            self._return_empty_list)
+
+        with travis_path_swap, mock_tests_to_remove:
+            with mock_e2e_travis_script:
+                with self.assertRaisesRegexp(Exception,
+                    'The e2e test suites that have been extracted from '
+                    'script section from travis.ci are empty.'):
+                        check_e2e_tests_are_captured_in_ci.main()
+
+    def test_main_with_invalid_protractor_test_suite_length(self):
+        protractor_path_swap = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_protractor_conf_file',
+            self._mock_read_protractor_conf_file)
+
+        mock_tests_to_remove = self._mock_tests_to_remove()
+
+        mock_e2e_test_suites = self.swap(
+            check_e2e_tests_are_captured_in_ci,
+            'get_e2e_suite_names_from_protractor_file',
+            self._return_empty_list)
+
+        with protractor_path_swap, mock_tests_to_remove:
+            with mock_e2e_test_suites:
+                with self.assertRaisesRegexp(Exception,
+                    'The e2e test suites that have been extracted from '
+                    'protractor.conf.js are empty.'):
+                        check_e2e_tests_are_captured_in_ci.main()
+
+    def test_main_without_errors(self):
+        protractor_path_swap = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_protractor_conf_file',
+            self._mock_read_protractor_conf_file)
+        travis_path_swap = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_and_parse_travis_yml_file',
+            self._mock_read_travis_yml_file)
+
+        mock_tests_to_remove = self._mock_tests_to_remove()
+
+        with protractor_path_swap, travis_path_swap, mock_tests_to_remove:
+            check_e2e_tests_are_captured_in_ci.main()
+
+
+TRAVIS_CI_DICT = {'env': {
+'jobs': [
+    'RUN_E2E_TESTS_ONEWORD=true', 'RUN_E2E_TESTS_TWO_WORDS=true']},
+'script': [
+    'if [ "RUN_E2E_TESTS_ONEWORD" == \'true\' ]; '
+    'then travis_retry bash scripts/run_e2e_tests.sh'
+    ' --suite="oneword" --prod_env; fi',
+    'if [ "RUN_E2E_TESTS_TWO_WORDS" == \'true\' ]; '
+    'then travis_retry bash scripts/run_e2e_tests.sh'
+    ' --suite="twoWords" --prod_env; fi']}
+
+DUMMY_PROTRACTOR_CONF_FILE = """var path = require('path')
+var suites = {
+  oneword: [
+    'protractor/oneword.js'
+  ],
+
+  twoWords: [
+    'protractor_desktop/twoWords.js'
+  ]
+};
+"""
```

Comment:
Missing newline at EOF. (In the future, you can avoid these errors by proofreading the Files Changed tab of your PR before asking for a review.)

Reply:
Done